pandas/io/excel/_base.py

```diff
@@ -1326,6 +1326,13 @@ def _value_with_fmt(
             fmt = "0"
         else:
             val = str(val)
+            # GH#56954
+            if len(val) > 32767:
+                warnings.warn(
```

Comment:
The issue mentioned warning, but we currently raise an error if the data has too many rows/columns. I think we should raise here too - I don't believe users would be okay with the data loss in certain cases.